query-connector/src/app/queryBuilding/buildFromTemplates/page.tsx

```diff
@@ -75,13 +80,28 @@ export default function QueryTemplateSelection() {
     };
   }, [selectedConditions, queryName]);
 
-  function handleCreateQueryClick(event: React.MouseEvent<HTMLButtonElement>) {
-    event.preventDefault();
-
-    validateForm();
-    if (!!queryName && !formError.queryName && !formError.selectedConditions) {
-      submitForm();
-    }
+  // ensures the fetchedConditions' checkbox statuses match
+  // the data in selectedCondtiions
+  function updateFetchedConditionIncludeStatus(
+    selectedConditions: CategoryNameToConditionOptionMap,
+  ) {
+    const prevFetch = structuredClone(fetchedConditions);
+    Object.entries(selectedConditions).map(
+      ([category, conditionsByCategory]) => {
+        Object.entries(conditionsByCategory).flatMap(
+          ([conditionId, conditionObj]) => {
+            if (prevFetch) {
```

Comment:
could we bump this check for `if(prevFetch)` up before the nested loop? Unless I'm missing something, we can just check for the definition for the fetched conditions once, which could save us a level of nesting here / might make this a bit easier to read